### PR TITLE
task threads ready lifecycle hook

### PR DESCRIPTION
**Pull request by duke8253**
Adding a new lifecycle hook `TS_LIFECYCLE_TASK_THREADS_READY_HOOK`.

This hook is invoked after `ET_TASK` threads started with event `TS_EVENT_LIFECYCLE_TASK_THREADS_READY`.

### CODE REVIEW

File: include/ts/apidefs.h.in [hunk 1]
```diff
@@ -324,7 +324,7 @@ typedef enum {
       accept connections. This is *not* guaranteed to be called
       before the first connection is accepted.
 
-      Event: TS_EVENT_LIFECYCLE_PORTS_READY_HOOK
+      Event: TS_EVENT_LIFECYCLE_PORTS_READY
```
Comment (shinrich) [inline include/ts/apidefs.h.in]: Did you mean to drop the _HOOK here?

File: doc/developer-guide/api/functions/TSLifecycleHookAdd.en.rst [hunk 2]
```diff
@@ -98,6 +98,13 @@ Types
 
       Called after every SSL context initialization used by |TS| for inbound connections (|TS| as the server).
 
+   .. cpp:enumerator:: TS_LIFECYCLE_TASK_THREADS_READY_HOOK
+
+      Called after |TS| task threads have been started.
+
+      Invoked with the event :c:data:`TS_EVENT_LIFECYCLE_TASK_THREADS_READY` and ``NULL``
+      data.
+
```
Comment (shinrich) [inline doc/developer-guide/api/functions/TSLifecycleHookAdd.en.rst]: Are the lifecycle hooks always synchronous?  If not, should put a comment that this plugin hook cannot be delayed.

File: include/ts/apidefs.h.in (hunk 1 above)
Comment (duke8253) [reply include/ts/apidefs.h.in]: yes, this is a typo as the event is here https://github.com/apache/trafficserver/blob/master/include/ts/apidefs.h.in#L452

File: doc/developer-guide/api/functions/TSLifecycleHookAdd.en.rst [hunk 3]
```diff
@@ -98,6 +98,13 @@ Types
 
       Called after every SSL context initialization used by |TS| for inbound connections (|TS| as the server).
 
+   .. cpp:enumerator:: TS_LIFECYCLE_TASK_THREADS_READY_HOOK
```
Comment (SolidWallOfCode) [inline doc/developer-guide/api/functions/TSLifecycleHookAdd.en.rst]: `cpp:enumerator`? Isn't this a C API?

File: src/traffic_server/traffic_server.cc [hunk 4]
```diff
@@ -1957,7 +1957,14 @@ main(int /* argc ATS_UNUSED */, const char **argv)
     RecConfigWarnIfUnregistered();
 
     // "Task" processor, possibly with its own set of task threads
-    tasksProcessor.start(num_task_threads, stacksize);
+    {
+      tasksProcessor.start(num_task_threads, stacksize);
```
Comment (SolidWallOfCode) [inline src/traffic_server/traffic_server.cc]: How confident are you that the task threads have actually started by the time the `start` method returns? I thought there was some sort of counter on the threads to track that.

**Pull request merged**